TypeError: this.db.setex is not a function

**Describe the bug**
```
...
[PROTOCOL] [6977169D-F902-4148-A1C9-DC946CA413FB.1.1] [outbound] S: 220 2.0.0 Ready to start TLS\r\n
[DEBUG] [-] [core] client TLS upgrade in progress, awaiting secured.
[ERROR] [-] [core] client TLS error: Error: 0058696FCB7F0000:error:0A000152:SSL routines:final_renegotiate:unsafe legacy renegotiation disabled:../deps/openssl/openssl/ssl/statem/extensions.c:922:
[ERROR] [6977169D-F902-4148-A1C9-DC946CA413FB.1.1] [outbound] Ongoing connection failed to -.-.-.-:25 : Error: 0058696FCB7F0000:error:0A000152:SSL routines:final_renegotiate:unsafe legacy renegotiation disabled:../deps/openssl/openssl/ssl/statem/extensions.c:922:
[DEBUG] [-] [core] [outbound] release_client: [39485294](https://bitbucket.org/analogic/mailserver/commits/39485294)-41EE-4237-AAE2-EBA2485B7193 -.-.-.-:25 to undefined
[NOTICE] [-] [OutboundTLS] TLS connection failed. Marking -.-.-.- as non-TLS for 604800 seconds
[CRIT] [-] [core] TypeError: this.db.setex is not a function
[CRIT] [-] [core] at OutboundTLS.mark_tls_nogo (/usr/lib/node_modules/Haraka/outbound/tls.js:98:17)
[CRIT] [-] [core] at pluggableStream.<anonymous> (/usr/lib/node_modules/Haraka/outbound/hmail.js:404:30)
[CRIT] [-] [core] at Object.onceWrapper (node:events:627:26)
[CRIT] [-] [core] at pluggableStream.emit (node:events:512:28)
[CRIT] [-] [core] at TLSSocket.<anonymous> (/usr/lib/node_modules/Haraka/tls_socket.js:90:18)
[CRIT] [-] [core] at Object.onceWrapper (node:events:627:26)
[CRIT] [-] [core] at TLSSocket.emit (node:events:524:35)
[CRIT] [-] [core] at emitErrorNT (node:internal/streams/destroy:151:8)
[CRIT] [-] [core] at emitErrorCloseNT (node:internal/streams/destroy:116:3)
[CRIT] [-] [core] at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
[NOTICE] [-] [core] Shutting down
```

I think the problem lies in the latest redis-client - I can't find the exact commit, but as I recall there is a slight change in the function naming, e.g. setex -> setEx. There may also be a change from cb-based results to async/await.

Haraka | Haraka.js — Version: 3.0.1
 --- | :--- 
Node | v19.6.0
OS | Linux mail.poste.io 4.19.0-22-amd64 #1 SMP Debian 4.19.260-1 (2022-09-29) x86_64 GNU/Linux
openssl | OpenSSL 1.1.1n  15 Mar 2022

--- a/Changes.md
+++ b/Changes.md
@@ -1,6 +1,9 @@
 
 ### Unreleased
 
+#### Fixed
+
+- fix: rename redis command setex to setEx #3174
 
 ### [3.0.1] - 2023-01-19
 

--- a/outbound/tls.js
+++ b/outbound/tls.js
@@ -95,7 +95,7 @@ class OutboundTLS {
 
         logger.lognotice(this, `TLS connection failed. Marking ${host} as non-TLS for ${expiry} seconds`);
 
-        this.db.setex(dbkey, expiry, (new Date()).toISOString())
+        this.db.setEx(dbkey, expiry, (new Date()).toISOString())
             .then(cb)
             .catch(err => {
                 logger.logerror(this, `Redis returned error: ${err}`);

